docker/python3-scientific/create_zarr_optimus.py

```diff
@@ -224,6 +225,40 @@ def add_cell_metrics(data_group, input_path, cell_ids, verbose=False):
         logging.info("Not adding \"cell_metadata_numeric\" to zarr output: either the #genes or # cell ids is 0")
 
 
+def create_gene_id_name_map(gtf_file):
+    """ Creates a map from gene_id to gene_name by reading in the GTF file
+
+    Args:
+        annotation_file (string): annotation file 
+
+    Return:
+        gene_id_name_map (dict[string]->string): dictonary gene ids to gene names
+    """
+    gene_id_name_map = {}
+
+    try:
+        if gtf_file.endswith(".gz"):
+            fpin = gzip.open(gtf_file, 'rt')
+        else:
+            fpin = open(gtf_file, 'r')
```

Comment:
I'm thinking if we use something like:
```python
with gzip.open(gtf_file, 'rt') if gtf_file.endswith(".gz") else open(gtf_file, 'r') as fpin:
    blah
```
we could get rid of the `fpin.close()` and make it "safer". Please test if it works if you decide to go this way since I didn't test it!

Reply:
Thanks! It worked!